contracts/NFTStorefront.cdc

```diff
@@ -448,6 +551,38 @@ pub contract NFTStorefront {
             return self.listings.keys
         }
 
+        // getDublicateListingIDs
+        // Returns an array of listing Ids that are duplicate for the given `nftType` and `nftId`.
+        //
+        pub fun getDuplicateListingIDs(nftType: Type, nftID: UInt64, listingID: UInt64): [UInt64] {
+            let listingIds = self.listedNFTs[nftType][nftID]
+            // Verify that given listing Id also a part of the `listingIds`
+            let isListingExisted = listingIds.contains(listingID)
+            if isListingExisted {
+                var index: Int = 0
+                for id in listingIds {
+                    if id == listingId {
+                        break
+                    }
+                    index = index + 1
+                }
+                return listingIds.remove(at:index)
+            } 
+            return []
+        }
+
+        // cleanupExpiredListings
+        // Cleanup the expired listing by iterating all the available listing
+        //
+        pub fun cleanupExpiredListings() {
```

Comment:
can this method contain a limit? otherwise it risks running out of gas and there's no way out of it

Reply:
True, I can add that. Thanks for feedback